### PR TITLE
(fix): Allow override of primary key type

### PR DESCRIPTION
**Pull request by JeSuisAlrick**
- Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by nicolas63** [comment]
Very nice work ! I will test it soon.  The sonar analysis is not a problem

File: generators/blazor/templates/src/client/Project.Client/Pages/Entities/_entityClass_/_entityClass_Detail.razor.ejs [hunk 1]
```diff
@@ -5,16 +5,16 @@ const entityClassName = pascalizedEntityClass;
 const entityVariableName = camelCasedEntityClass;
 const entityClassNamePluralized = pascalizedEntityClassPlural;
 _%>
-@page "/<%= lowerCasedEntityClass %>/{Id<%_ if(primaryKeyType !== 'string') { %>:<%= primaryKeyType %> <%_ } _%> }/view"
+@page "/<%= lowerCasedEntityClass %>/{Id<%_ if(isNumericPrimaryKey(primaryKeyType)) { %>:<%= primaryKeyType %> <%_ } _%> }/view"
```
Comment (nicolas63) [inline generators/blazor/templates/src/client/Project.Client/Pages/Entities/_entityClass_/_entityClass_Detail.razor.ejs:8]: here we need the id type like "{Id:guid}" without this the detail page can't convert the id

File: generators/blazor/templates/src/client/Project.Client/Pages/Entities/_entityClass_/_entityClass_Update.razor.ejs [hunk 2]
```diff
@@ -6,7 +6,7 @@ const entityVariableName = camelCasedEntityClass;
 const entityClassNamePluralized = pascalizedEntityClassPlural;
 _%>
 @page "/<%= lowerCasedEntityClass %>/new"
-@page "/<%= lowerCasedEntityClass %>/{Id<%_ if(primaryKeyType !== 'string') { %>:<%= primaryKeyType  %> <%_ } _%> }/edit"
+@page "/<%= lowerCasedEntityClass %>/{Id<%_ if(isNumericPrimaryKey(primaryKeyType)) { %>:<%= primaryKeyType  %> <%_ } _%> }/edit"
```
Comment (nicolas63) [inline generators/blazor/templates/src/client/Project.Client/Pages/Entities/_entityClass_/_entityClass_Update.razor.ejs:9]: Like the detailt page we need the id type here

**Review by nicolas63 — CHANGES_REQUESTED** [review]

**Review by nicolas63 — APPROVED** [review]
Everything looks good, if you agree, I can merge it.

**Comment by JeSuisAlrick** [comment]
> Everything looks good, if you agree, I can merge it.

Yeah. I'm good with the changes.

I'm considering migrations as the next thing to work on, but I'll support this change.

**Comment by nicolas63** [comment]
Merged, thanks for your contribution

**Pull request merged**